fdbclient/NativeAPI.actor.cpp

```diff
@@ -5344,9 +5354,9 @@ void Transaction::checkDeferredError() {
 
 Reference<TransactionLogInfo> Transaction::createTrLogInfoProbabilistically(const Database& cx) {
 	if (!cx->isError()) {
-		double clientSamplingProbability = std::isinf(cx->clientInfo->get().clientTxnInfoSampleRate)
-		                                       ? CLIENT_KNOBS->CSI_SAMPLING_PROBABILITY
-		                                       : cx->clientInfo->get().clientTxnInfoSampleRate;
+		double sampleRate = GlobalConfig::globalConfig().get<double>(fdbClientInfoTxnSampleRate,
+		                                                             std::numeric_limits<double>::infinity());
```

Comment:
Is it possible for us to simplify this by passing in the client knob as a default? That changes the behavior if the actual key is set to infinity, but if we're moving the key maybe we can make that something that doesn't happen.